pkg/parse/run.go

```diff
@@ -238,6 +251,14 @@ func run(ctx context.Context, p Parser, trigger string, state *reconcilerState)
 // It also updates the .status.rendering and .status.source fields.
 func read(ctx context.Context, p Parser, trigger string, state *reconcilerState, sourceState sourceState) status.MultiError {
 	hydrationStatus, sourceStatus := readFromSource(ctx, p, trigger, state, sourceState)
+	if p.options().renderingEnabled != hydrationStatus.requiresRendering {
+		// the reconciler is misconfigured. set the annotation so that the reconciler-manager
+		// will recreate this reconciler with the correct configuration.
```

Comment:
Why do we need both `status.requiresRendering` and the requiresRendering annotation? Can we just keep one?

Reply:
There is no `status.requiresRendering`, at least on the `RSync`. This status is just used internally to compare the rendering status in the SoT to the current configuration/annotation